coordinator/coordinator.go

```diff
@@ -87,12 +87,19 @@ func (c *Coordinator) Run() (<-chan struct{}, error) {
 
 		defer func() {
 			c.monitor.Stop()
+			clusterStatus, err := c.cluster.Status()
+			if err != nil {
+				c.log.Error("coordinator: cluster status error:", mlog.Err(err))
+			}
 			c.cluster.Shutdown()
 			close(c.doneChan)
 			c.mut.Lock()
 			c.status.State = Done
 			c.status.SupportedUsers = supported
 			c.status.StopTime = time.Now()
+			if clusterStatus.NumErrors > 0 {
```

Comment:
Is this check even needed?

Reply:
I've added it because `cluster.Status()` doesn't really return an error but in case of one it can return a zero status value. I wanted to avoid a case where we might be overwriting an existing value for `c.status.NumErrors`.